Salesforce/client/http_client.py

```diff
@@ -0,0 +1,272 @@
+        https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/event_log_file_hourly_overview.htm
+
+        Value of filter criterion for field 'CreatedDate' must be of type dateTime and should not be enclosed in quotes.
+
+        Args:
+            start_from: str | None
+
+        Returns:
+            str:
+        """
+        date_filter = "AND CreatedDate > {0}".format(start_from) if start_from else ""
+
+        return """
+            SELECT Id, EventType, LogFile, LogDate, LogFileLength
+                FROM EventLogFile WHERE Interval = \'Hourly\' {0}
+        """.format(
+            date_filter
+        )
+
+    def _request_url_with_query(self, query: str) -> URL:
+        """
+        Creates request url to perform query operation.
+
+        Include query formatting and correct encoding.
+
+        Args:
+            query: str
+
+        Returns:
+            URL:
+        """
+        params = {
+            "q": self._query_to_http_param(query),
+        }
+
+        return URL("{0}/services/data/v58.0/query".format(self.base_url)).with_query(
+            urlencode(params, safe=">+,'=:", encoding="utf-8")
+        )
+
+    async def _request_headers(self) -> Dict[str, str]:
+        """
+        Get request headers with auth.
+
+        Returns:
+            Dict[str, str]:
+        """
+        token_refresher = await self._get_token_refresher()
+
+        async with token_refresher.with_access_token() as token:
+            headers = {
+                "Authorization": "Bearer {0}".format(token.token.access_token),
+                "Content-Type": "application/json",
+            }
+
+        return headers
+
+    @classmethod
+    async def _handle_error_response(cls, response: ClientResponse) -> None:
+        """
+        Handle error response.
+
+        Args:
+            response: ClientResponse
+
+        Raises:
+            Exception: in case if response status is not 200
+        """
+        if response.status != 200:
+            try:
+                error_response = await response.json()
+            except Exception:
+                error_msg = await response.text()
+
+                raise Exception(error_msg)
+
+            error_msg = ",".join(["Error {errorCode}: {message}".format(**error) for error in error_response])
+
+            raise Exception(error_msg)
+
+    async def get_log_files(self, start_from: str | None = None) -> SalesforceEventLogFilesResponse:
+        """
+        Get log files from Salesforce.
+
+        Args:
+            start_from: str
+
+        Raises:
+            ValueError: Salesforce response cannot be processed
+
+        Returns:
+            SalesforceEventLogFilesResponse:
+        """
+        logger.info("Getting log files from Salesforce. Start date is {0}", start_from)
+
+        query = self._log_files_query(start_from)
+
+        url = self._request_url_with_query(query)
+        headers = await self._request_headers()
```

Comment:
Cannot we add the `Accept-Encoding: gzip` header to manage heavy responses? ([see](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/intro_rest_compression.htm#intro_rest_compression))